### DISCUSSION TITLE
Accessing search params via beforeLoad

### DISCUSSION THREAD

**Discussion by shansmith01** [TanStack/router#800] [Q&A] [answered]
I understnad that we should be able to access search params from beforeLoad.

If I do this...

`validateSearch: (search: Record<string, unknown>): SitesSearchParams => {
    return { token: search.token as string }
  },
  beforeLoad: async (all) => {
    console.log(all)
  },`

And check my console all I see is

`{abortController: AbortController, params: {…}, preload: false, context: {…}}
abortController
: 
AbortController {signal: AbortSignal}
context
: 
{getSite: ƒ}
params
: 
{site_id: '1234'}
preload
: 
false
[[Prototype]]
: 
Object`

Noting about search params. Can anyone help me access the search object?

**Comment by tannerlinsley** [comment, marked as answer]
Make sure you’re on the latest.

Tanner Linsley
